muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -211,7 +263,18 @@ public Response toggleArchived(@PathParam ("NOTEID") Long noteId) {
     }
 
     Note updatedNote = notesController.toggleArchived(note);
+    if(note.getArchived().equals(Boolean.FALSE)) {
+      note.setDueDate(null);
+      note.setStartDate(null);
+    }
     
-    return Response.ok(toRestModel(updatedNote)).build();
+    NoteRestModel updatedRestModel = toRestModel(updatedNote);
+    updatedRestModel.setIsActive(true);
+
+    if (note.getArchived().equals(Boolean.TRUE)) {
```

Comment:
Is this certainly `note` instead of `updatedNote` (could be fine - just a bit hard to read what exactly is going on).